### DISCUSSION TITLE
Update of textual to 0.86?

### DISCUSSION THREAD

**Discussion by pcboy** [tconbeer/harlequin#712] [General]
I'm the maintainer of the Harlequin package on NixOS.  
And it happens that currently the harlequin package can't run anymore as the `textual` dependency has been updated to 0.86 on master branch of NixOS.

https://github.com/NixOS/nixpkgs/issues/364867

We get this error on first run:

```
  File "/nix/store/hz7m3hlp8m2drds2m45za58vzk5pqi4y-python3.12-textual-0.86.1/lib/python3.12/site-packages/textual/app.py", line 1249, in _validate_theme
    raise InvalidThemeError(message)
textual.app.InvalidThemeError: Theme 'harlequin' has not been registered. Call 'App.register_theme' before setting the 'App.theme' attribute.
```

Is there any plan to update to 0.86 any time soon by any chance? Or any idea of a quick fix for this situation?

**Comment by tconbeer** [comment]
Yeah, there are a lot of breaking changes in that release. Fortunately 1.0 just dropped and that will hopefully be the end of it for a while. 

I have a branch with a fix, but it isn't finished yet. I decided to drop Harlequin's themes entirely in favor of Textual's new themes, which makes breaking changes in user land and likely Harlequin 2.0. I need to finish up the tests and write new docs. Between work and the holidays I'm not sure when that will get done, tbh... hopefully this year. 

I'm tracking over at https://github.com/tconbeer/harlequin/issues/697

**Reply by pcboy** [reply]
I see, okay. Thank for the quick response!

It's good to know that there is a plan to support the new textual version. 
Will wait for the fix then and will update the nix package to the new version when available.

**Reply by tconbeer** [reply]
v2.0.0 has been released, which should fix your problems. Sorry for the delay -- this was a bit of a bear. Let me know how it goes.

**Reply by pcboy** [reply]
Thanks a lot, seems to be working fine on my side, opened a PR on nixpkgs to update the package. :+1: 

https://github.com/NixOS/nixpkgs/pull/376079
